manifests/init.pp

```diff
@@ -376,12 +376,27 @@
     if $agent_major_version != undef and $agent_major_version != $_agent_major_version {
       fail('Provided and deduced agent_major_version don\'t match')
     }
+    $_agent_minor_version = 0 + $3
   } elsif $agent_major_version != undef {
     $_agent_major_version = $agent_major_version
   } else {
     $_agent_major_version = $datadog_agent::params::default_agent_major_version
   }
 
+  case $facts['os']['name'] {
+    'RedHat', 'CentOS', 'OracleLinux': {
+      if $facts['os']['release']['full'] =~ /^6(.[0-9])?/ and $agent_version == 'latest' {
+        notice("datadog-agent ${_agent_major_version}.51 is the last supported version on CentOS 6. Installing ${_agent_major_version}.51 now")
+        $agent_full_version='7.51.1'
```

Comment:
Would something similar to this work? (though at this point it might be unlikely we'll release more patches for 7.51)
```suggestion
        $agent_full_version='< 1:7.52.0-1'
```

Reply:
At first I tried a similar format with `1:7.51.1-1` and it didn't work. I think I may need to move a few things around to make it work, I'll try in a separate PR.